### ISSUE TITLE
[Bug]: Assorted accessibility issues on AoT 2024 editor

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typehero/typehero#1686]
### What happened?

1. Visit https://www.adventofts.com/events/2024/1
2. In a Monaco code editor, press <kbd>F1</kbd>, type something like `high contrast`, and click the result or hit <kbd>Enter</kbd> to enable Monaco's high contrast theme[^monaco-a11y]
3. Run Lighthouse's Accessibility check (i.e. aXe)
4. See complaints:
    1. [Buttons do not have an accessible name](https://dequeuniversity.com/rules/axe/4.10/button-name)
    2. [Background and foreground colors do not have sufficient color contrast](https://dequeuniversity.com/rules/axe/4.10/color-contrast)

### What browsers are you seeing the problem on?

All

### Relevant ~~log output~~ info

When text doesn't have enough color contrast against its background, folks with limited vision -such as those with bad eyesight or lots of screen glare- might not be able to read the text.

When buttons or links don't have an accessible name, folks who can't visually see the button won't have any way of knowing what they are.

[^monaco-a11y]: Monaco has a rich accessibility story, courtesy of Microsoft needing to make VS Code accessible: https://github.com/Microsoft/monaco-editor/wiki/Monaco-Editor-Accessibility-Guide.
